packages/apps/plugins/plugin-stack/src/components/AddSectionDialog.tsx

```diff
@@ -0,0 +1,90 @@
+//
+// Copyright 2024 DXOS.org
+//
+import React, { useCallback, useState } from 'react';
+
+import { Folder, Stack as StackType } from '@braneframe/types';
+import { usePlugin, useIntent, LayoutAction } from '@dxos/app-framework';
+import { getSpaceForObject, useQuery } from '@dxos/react-client/echo';
+import { Dialog, toLocalizedString, useTranslation } from '@dxos/react-ui';
+import { Path } from '@dxos/react-ui-mosaic';
+import { SearchList } from '@dxos/react-ui-searchlist';
+import { type AddSectionPosition } from '@dxos/react-ui-stack';
+import { getSize, mx } from '@dxos/react-ui-theme';
+
+import { STACK_PLUGIN } from '../meta';
+import { type StackPluginProvides } from '../types';
+
+type AddSectionDialogProps = { path?: string; position: AddSectionPosition; stack: StackType };
+
+export const dataHasAddSectionDialogProps = (data: any): data is { subject: AddSectionDialogProps } => {
```

Comment:
is this used?

Reply:
Yes, it is the guard used by the component to give a type to `data`.

Reply:
oh I see it now 👍